### PR TITLE
TOF: extend digit QC

### PR DESCRIPTION
**Pull request by njacazio**
- Extend configuration of checks/tasks
- Use standard types
- Improve performance
- Use std::make_shared
- Using default destructor in TaskDigits
- Add running mode for TaskDigits
- Simplified RawMultiplicity checker
- Improve messages for shifter
- Update TOF jsons

### CODE REVIEW

File: Modules/TOF/include/Base/MessagePad.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+// ROOT includes
+#include "TPaveText.h"
+#include "TList.h"
+#include "TH1F.h"
+#include "TH2F.h"
+
+namespace o2::quality_control_modules::tof
+{
+
+struct MessagePad {
```
Comment (ercolessi) [inline Modules/TOF/include/Base/MessagePad.h:31]: Great idea

**Comment by njacazio** [comment]
@Barthelemy @knopers8 we are ready to have this reviewed and merged! Suggestions are as always more than welcome, note that I added a class to have a message pad in the QCG, maybe this could be a framework feature instead of a TOF one.
Thanks!

File: Modules/TOF/include/Base/MessagePad.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  float mPadLowX = 0.6;                 /// Position of the message PAD in low x
+  float mPadLowY = 0.5;                 /// Position of the message PAD in low y
+  float mPadHighX = 0.9;                /// Position of the message PAD in high x
+  float mPadHighY = 0.75;               /// Position of the message PAD in high y
+  std::vector<std::string> mMessages{}; /// Message to print
+  int mEnabledFlag = 1;                 /// Flag to enable or disable the pad
+
+  template <typename T>
+  void Configure(const T& CustomParameters)
```
Comment (Barthelemy) [inline Modules/TOF/include/Base/MessagePad.h:43]: We use lower-case initial letter in O2.

File: Modules/TOF/src/CheckCompressedData.cxx [hunk 3]
```diff
@@ -70,34 +67,15 @@ void CheckCompressedData::beautify(std::shared_ptr<MonitorObject> mo, Quality ch
...
     auto* h = dynamic_cast<TH2F*>(mo->getObject());
-    TPaveText* msg = new TPaveText(0.9, 0.1, 1.0, 0.5, "blNDC");
-    h->GetListOfFunctions()->Add(msg);
-    msg->SetBorderSize(1);
-    msg->SetTextColor(kWhite);
-    msg->SetFillColor(kBlack);
-    msg->AddText("Default message for hDiagnostic");
-    msg->SetName(Form("%s_msg", mo->GetName()));
-
+    TPaveText* msg = mShifterMessages.MakeMessagePad(h, checkResult);
```
Comment (Barthelemy) [inline Modules/TOF/src/CheckCompressedData.cxx:70]: I think that there is a memory leak here. `msg` is not deleted.

File: Modules/TOF/include/Base/MessagePad.h [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  void AddMessage(const std::string& message)
+  {
+    mMessages.push_back(message);
+  }
+
+  template <typename T>
+  TPaveText* MakeMessagePad(T* histogram, const Quality& quality)
+  {
+    TPaveText* msg = new TPaveText(mPadLowX, mPadLowY, mPadHighX, mPadHighY, "blNDC");
```
Comment (Barthelemy) [inline Modules/TOF/include/Base/MessagePad.h:78]: You could use a smart pointer here to avoid problems with memory leaks.

**Pull request merged**